### DISCUSSION TITLE
Logical IDs changed in all my stacks when migrating from v1 to v2, why is that?

### DISCUSSION THREAD

**Discussion by yagomassiah** [aws/aws-cdk#29885] [Q&A]
Hello folks, I've been running into some problems with my migration from cdk V1 to v2, I followed a few guides and was able to port most of the project, it came down to changing imports to use from `'aws-cdk-lib'` and a few other minor changes, but in the end I was not expecting `cdk diff` to output differences for ALL my stacks, basically from the looks of it logical IDs changed and now I'm required to destroy all my stacks and recreate them? that sounds a bit off, because in the docs they say logical ids shouldn't change.

For a bit more context I'm running a project currently sitting at `1.204` 

Here's an example on how some of these logical ids changed

```
 [ ]   "Resource": {
            [-]     "Fn::ImportValue": "SecretsStack:SecretsStackExportsOutputRefrelationalDbUser518518343FF79715"
            [+]     "Fn::ImportValue": "SecretsStack:ExportsOutputRefrelationalDbUser518518342706F1
```

**Comment by rgoltz** [comment]
I just remember some edge cases related to some resources. There is a pattern for dedicated resources getting imported?

A (bloody) workaround could be to set the logicalId: https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_core.CfnResource.html#overridewbrlogicalwbridnewlogicalid
